src/components/CreatePost/CreatePost.tsx

```diff
@@ -49,64 +54,222 @@ const CreatePost = () => {
     }));
   };
 
-  const firestore = useContext(FirestoreContext);
+  const { firestore, storage } = useContext(FirebaseContext);
+
+  const [title, setTitle] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [selectedFiles, setSelectedFiles] = useState<FilesInterface[]>([]);
+  const [selectedImages, setSelectedImages] = useState<ImagesInterface[]>([]);
+  const [showForm, setShowForm] = useState<boolean>(true);
+  const [titleChars, setTitleChars] = useState<number>(maxTitleChars);
+  const [descrChars, setDescrChars] = useState<number>(maxDescrChars);
+
+  const fileInput = React.createRef<HTMLInputElement>();
+
+  const storeImages = async () => {
+    let urls: string[] = [];
+    if (selectedFiles) {
+      for (let i = 0; i < selectedFiles.length; i++) {
+        const childRef = storage.child(selectedFiles[i].file.name);
+        const snapshot = await childRef.put(selectedFiles[i].file);
+        const url = await snapshot.ref.getDownloadURL();
+        urls = [...urls, url];
+        console.log(
+          `i: ${i}, id: ${selectedFiles[i].id}, file: ${selectedFiles[i].file.name}, url: ${url}`,
+        );
+      }
+      console.log('done!');
+    }
+    return urls;
+  };
+
+  const handleFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const savedRoles = Object.keys(selectedRoles).filter(
+      (key) => selectedRoles[key] === true,
+    );
+
+    try {
+      const urls = await storeImages();
+
+      const doc = await firestore.collection('projects').add({
+        title,
+        description,
+        roles: savedRoles,
+        images: urls,
+      });
+      console.log(`saved doc with id: ${doc.id}`);
+    } catch (err) {
+      console.log(`error saving doc: ${err}`);
+    }
+    setTitle('');
+    setDescription('');
+    setSelectedFiles([]);
+    setSelectedImages([]);
+    setSelectedRoles(defaultSelectedRoles);
+    console.log('DONE!');
+  };
+
+  const handleFileUpload = async () => {
+    if (fileInput.current) {
+      fileInput.current.click();
+    }
+  };
+
+  const handleFileSelected = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inFiles = e.target.files;
+    if (inFiles) {
+      Array.from(inFiles).map((file) => {
+        const id = Math.random().toString(36).substr(2, 9);
```

Comment:
We should import the uuid library and use `uuid.v4()` to generate a random id. Also not sure why we are generating ids during selection of images.